File: backend/app/constants/board_workflow.py
from app.db.models import TaskStatus

DEFAULT_BOARD_LISTS: list[tuple[str, TaskStatus]] = [
    ("Backlog", TaskStatus.BACKLOG),
    ("Blocked", TaskStatus.BLOCKED),
    ("In Progress", TaskStatus.IN_PROGRESS),
    ("In Review", TaskStatus.IN_REVIEW),
    ("QA", TaskStatus.QA),
    ("Done", TaskStatus.DONE),
]

LIST_NAME_TO_STATUS: dict[str, TaskStatus] = {name: status for name, status in DEFAULT_BOARD_LISTS}

STATUS_TO_LIST_NAME: dict[TaskStatus, str] = {status: name for name, status in DEFAULT_BOARD_LISTS}


def status_for_list_name(name: str) -> TaskStatus | None:
    return LIST_NAME_TO_STATUS.get(name)


def list_name_for_status(status: TaskStatus) -> str | None:
    return STATUS_TO_LIST_NAME.get(status)

File: frontend/src/entities/boardWorkflow.ts
import type { TaskStatus } from "./taskFields";

export const DEFAULT_BOARD_LISTS: { name: string; status: TaskStatus }[] = [
  { name: "Backlog", status: "backlog" },
  { name: "Blocked", status: "blocked" },
  { name: "In Progress", status: "in_progress" },
  { name: "In Review", status: "in_review" },
  { name: "QA", status: "qa" },
  { name: "Done", status: "done" },
];

const LIST_NAME_TO_STATUS = Object.fromEntries(
  DEFAULT_BOARD_LISTS.map(({ name, status }) => [name, status])
) as Record<string, TaskStatus>;

const STATUS_TO_LIST_NAME = Object.fromEntries(
  DEFAULT_BOARD_LISTS.map(({ name, status }) => [status, name])
) as Record<TaskStatus, string>;

export function statusForListName(name: string): TaskStatus | null {
  return LIST_NAME_TO_STATUS[name] ?? null;
}

export function listNameForStatus(status: TaskStatus): string | null {
  return STATUS_TO_LIST_NAME[status] ?? null;
}

export function statusForListId(
  listId: string | undefined,
  lists: { id: string; name: string }[] | undefined
): TaskStatus {
  if (!listId || !lists) return "backlog";
  const list = lists.find((item) => item.id === listId);
  return list ? statusForListName(list.name) ?? "backlog" : "backlog";
}
